### PR TITLE
Add podman kube generate test

### PR DESCRIPTION
**Pull request by ilausuch**
- Related ticket: https://progress.opensuse.org/issues/124601
- Verification run: 
- 15-SP4 x86  http://openqa.suse.de/tests/10826614
- 15-SP4 aarch64 http://openqa.suse.de/tests/10826695
- SP3 http://openqa.suse.de/tests/10827390
- SP2 http://openqa.suse.de/tests/10826718 (Excluded)
- SP1 http://openqa.suse.de/tests/10826719 (Excluded)
- Microos https://openqa.opensuse.org/tests/3200047
- TW https://openqa.opensuse.org/tests/3200048

### CODE REVIEW

File: tests/containers/podman_pods.pm [hunk 1]
```diff
@@ -41,6 +41,12 @@ sub run {
         record_info('Cleanup', 'Stop pods');
         assert_script_run('podman play kube --down hello-kubic.yaml');
     }
+
+    assert_script_run('podman run -d --name generate_test registry.suse.com/bci/python:latest python3 --version');
+    my $pod_id = script_output('podman ps -a | grep generate_test | cut -d " " -f1');
```
Comment (pdostal) [inline tests/containers/podman_pods.pm:46]:
```suggestion
    my $container_id = script_output('podman container inspect generate_test --format "{{.Id}}"');
```

**Review by pdostal — CHANGES_REQUESTED** [review]
Maybe you can also save the yaml, remove the container and apply the yaml and then run the container so we're sure it's valid?

Also maybe you can also actually create a pod and place the container in it? E.g:
```bash
podman pod create testing_pod
podman container create --pod testing_pod --name generate_test …
```
But I'm not forcing you of course, that are just ideas.

File: tests/containers/podman_pods.pm (hunk 1 above)
Comment (ilausuch) [reply tests/containers/podman_pods.pm:46]: Thanks are good ideas

Comment (ilausuch) [reply tests/containers/podman_pods.pm]: done

**Comment by ilausuch** [comment]
I implemented your ideas. Thank you

**Review by pdostal — APPROVED** [review]
Awesome 🤩

File: tests/containers/podman_pods.pm [hunk 2]
```diff
@@ -41,6 +41,24 @@ sub run {
...
+    assert_script_run('podman container create --pod testing_pod --name container registry.suse.com/bci/python:latest bash -c "while true; do sleep 1; done"');
+    assert_script_run("podman kube generate testing_pod > pod.yaml");
+    assert_script_run("grep 'image: registry.suse.com/bci/python:latest' pod.yaml");
+    assert_script_run("podman pod rm testing_pod");
+
+    record_info('Test', 'Test the pod yaml creates a pod');
+    assert_script_run('podman play kube pod.yaml');
+    record_info('Test', 'Confirm pod is running');
+    record_info('pod ps', script_output('podman pod ps'));
+    assert_script_run('podman pod ps | grep "testing_pod"');
```
Comment (ilausuch) [inline tests/containers/podman_pods.pm:56]: I will change grep by validate_script_output

**Review by ricardobranco777 — APPROVED** [review]
LGTM

**Review by jlausuch — COMMENTED** [review]
Looks good! Maybe we could use `script_retry` to pull the image just in case?

So, if I understand correctly , `podman kube play pod.yaml` and `podman play kube pod.yaml` does the same thing?

File: tests/containers/podman_pods.pm [hunk 3]
```diff
@@ -40,6 +40,39 @@ sub run {
...
+        assert_script_run("podman pod rm testing_pod");
+
+        record_info('Test', 'Test the pod yaml creates a pod');
+        assert_script_run('podman play kube pod.yaml');
+        record_info('Test', 'Confirm pod is running');
+        record_info('pod ps', script_output('podman pod ps'));
+        validate_script_output('podman pod ps', sub { m/testing_pod/ });
+        validate_script_output('podman ps', sub { m/testing_pod-container/ });
+
+        record_info('Test', 'Killing one pod');
```
Comment (pdostal) [inline tests/containers/podman_pods.pm:60]:
```suggestion
        record_info('Test', 'Removing one pod');
```

**Review by pdostal — COMMENTED** [review]
Is that intentional to heve `podman kube plan` in one block and `podman play kube` in another? Maybe you can somehow describe it in `record_info()`

File: tests/containers/podman_pods.pm [hunk 4]
```diff
@@ -40,6 +40,39 @@ sub run {
         record_info('Cleanup', 'Stop pods');
         assert_script_run('podman play kube --down hello-kubic.yaml');
     }
+
+    unless (is_sle('<=15-SP2')) {
+        # Kube generate
+        record_info('Test', 'Generate the yaml from a pod');
+        assert_script_run('podman pod create testing_pod');
+        assert_script_run('podman container create --pod testing_pod --name container registry.suse.com/bci/python:latest bash -c "while true; do sleep 1; done"');
```
[the remark this answers is not in the document]
Comment (ilausuch) [reply tests/containers/podman_pods.pm:48]: good point, thanks!

**Comment by ilausuch** [comment]
Thanks for the reviews

File: tests/containers/podman_pods.pm (hunk 4 above)
[the remark this answers is not in the document]
Comment (pdostal) [reply tests/containers/podman_pods.pm:48]:
```suggestion
        my $image = 'registry.suse.com/bci/python:latest';
        script_retry "podman image $image";
        assert_script_run("podman container create --pod testing_pod --name container $image bash -c 'while true; do sleep 1; done'");
```

**Review by ricardobranco777 — CHANGES_REQUESTED** [review]
We could use `bci-busybox` instead of `python:3.10` as the image is much shorter:

```
$ podman images
REPOSITORY                           TAG         IMAGE ID      CREATED       SIZE
registry.suse.com/bci/bci-busybox    latest      8479bde8aeb3  5 weeks ago   14.4 MB
registry.suse.com/bci/python         3.10        45eb3f9210d9  5 weeks ago   235 MB
```

File: tests/containers/podman_pods.pm [hunk 5]
```diff
@@ -40,6 +40,41 @@ sub run {
...
         assert_script_run('podman play kube --down hello-kubic.yaml');
     }
+
+    unless (is_sle('<=15-SP2')) {
+        # Kube generate
+        record_info('Test', 'Generate the yaml from a pod');
+        assert_script_run('podman pod create testing_pod');
+        image = "registry.suse.com/bci/bci-busybox:latest";
+        script_retry("podman pull -q $image", timeout => 300, delay => 60, retry => 3);
+        assert_script_run("podman container create --pod testing_pod --name container image sh -c \"while true; do sleep 3600; done\"");
```
Comment (ricardobranco777) [inline tests/containers/podman_pods.pm:50]:
```suggestion
        assert_script_run("podman container create --pod testing_pod --name container $image sleep 3600");
```
You forgot the `$` :)

Also, I think we can drop the `while` loop and just use `sleep 3600` as we don't expect the container to run that longer.

**Review by ricardobranco777 — CHANGES_REQUESTED** [review]

**Comment by ilausuch** [comment]
I am testing yet.. I will notify when is ready again :)

File: tests/containers/podman_pods.pm [hunk 6]
```diff
@@ -40,6 +41,41 @@ sub run {
...
         assert_script_run('podman play kube --down hello-kubic.yaml');
     }
+
+    unless (is_sle('<=15-SP2')) {
+        # Kube generate
+        record_info('Test', 'Generate the yaml from a pod');
+        assert_script_run('podman pod create testing_pod');
+        my $image = "registry.suse.com/bci/bci-busybox:latest";
+        script_retry("podman pull $image", timeout => 300, delay => 60, retry => 3);
+        assert_script_run("podman container create --pod testing_pod --name container $image sh -c \"sleep 3600\"");
```
Comment (ricardobranco777) [inline tests/containers/podman_pods.pm:51]:
```suggestion
        assert_script_run("podman container create --pod testing_pod --name container $image sleep 3600");
```
You don't really need the `sh` to run `sleep` :)

**Review by ricardobranco777 — APPROVED** [review]
Just a small suggestion. Otherwise LGTM

**Review by jlausuch — APPROVED** [review]

**Pull request merged**